addons/binding/org.openhab.binding.chromecast/src/main/java/org/openhab/binding/chromecast/internal/ChromecastScheduler.java

```diff
@@ -33,40 +34,43 @@
     private ScheduledFuture<?> connectFuture;
     private ScheduledFuture<?> refreshFuture;
 
-    public ChromecastScheduler(ScheduledExecutorService scheduler, Integer connectDelay, Runnable connectRunnable, Integer refreshRate, Runnable refreshRunnable) {
+    public ChromecastScheduler(ScheduledExecutorService scheduler, Integer connectDelay, Runnable connectRunnable,
+            Integer refreshRate, Runnable refreshRunnable) {
         this.scheduler = scheduler;
         this.connectDelay = connectDelay;
         this.connectRunnable = connectRunnable;
         this.refreshRate = refreshRate;
         this.refreshRunnable = refreshRunnable;
     }
 
-    public void destroy() {
+    public synchronized void destroy() {
         cancelConnect();
         cancelRefresh();
     }
 
-    public void scheduleConnect() {
+    public synchronized void scheduleConnect() {
         logger.debug("Scheduling connection");
         cancelConnect();
         connectFuture = scheduler.schedule(connectRunnable, connectDelay, SECONDS);
     }
 
-    private void cancelConnect() {
+    private synchronized void cancelConnect() {
         logger.debug("Canceling connection");
         if (connectFuture != null) {
             connectFuture.cancel(true);
             connectFuture = null;
         }
     }
 
-    public void scheduleRefresh() {
+    public synchronized void scheduleRefresh() {
         cancelRefresh();
         logger.debug("Scheduling refresh in {} seconds", refreshRate);
-        refreshFuture = scheduler.scheduleAtFixedRate(refreshRunnable, 0, refreshRate, SECONDS);
+        // With an initial delay of 1 second the refresh job can be restarted when several channels
+        // are refreshed at once e.g. due to channel linking
+        refreshFuture = scheduler.scheduleAtFixedRate(refreshRunnable, 1, refreshRate, SECONDS);
```

Comment:
Please use [scheduleWithFixedDelay](http://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ScheduledExecutorService.html#scheduleWithFixedDelay(java.lang.Runnable,%20long,%20long,%20java.util.concurrent.TimeUnit)) which is preferred over scheduleAtFixedRate, see [Coding Guidelines D.3](http://docs.openhab.org/developers/development/guidelines.html#d-runtime-behavior)

Reply:
Fixed! I still wanted to check the Git history before changing it. :-) I couldn't find any reason why scheduleAtFixedRate was used.